### ISSUE TITLE
Improve `print(mesh)`

### ISSUE THREAD

**Issue by prisae** [simpeg/discretize#142]
I think the current print-utility of meshes was made with small meshes in mind. It sort of explodes for big, stretched meshes:

![screenshot](https://user-images.githubusercontent.com/8020943/54692397-3ebda580-4b25-11e9-989e-1c93f1a68bb2.png)

I think something along these lines would be preferable (maybe as a rendered table in notebooks):

    discretize.TensorMesh: 128 x 128 x 128 cells; 2,097,152 in total
    
        dir          min/max N            min/max h
        -----------------------------------------------------
         x:      -4176.3 / 4176.3;     25.0 / 143.0
         y:      -5902.6 / 5902.6;     50.0 / 160.8
         z:      -7151.7 / 7151.7;     20.0 / 355.8

**Comment by prisae** [comment]
Related, to some extent, to #21.

**Comment by banesullivan** [comment]
I could help implement a  `_repr_html_` method like what I did for `vtki`:

<img width="730" alt="Screen Shot 2019-03-20 at 10 20 54 AM" src="https://user-images.githubusercontent.com/22067021/54701087-e2915c00-4af9-11e9-8eb9-483b92455cea.png">

**Comment by prisae** [comment]
I think that would be great! But I think @lheagy and @rowanc1 and probably others of SimPEG have to chime in with their opinion, as it is their brainchild.

A few points:
- What happens if you are not in a notebook? Is there a plain text fallback?
- I think minimum cell-width in each distance would be useful.
- Maybe format the bounds as `.1f` instead of `.3e`

**Comment by banesullivan** [comment]
The `_repr_html_` method is only ever used in an IPython environment. If you want a plain text representation to fall back on, then it can be implemented in the `__repr__` method:

<img width="504" alt="Screen Shot 2019-03-20 at 12 39 09 PM" src="https://user-images.githubusercontent.com/22067021/54710390-2e013580-4b0d-11e9-90be-eb1c037d5370.png">

**Comment by prisae** [comment]
It would also go nicely along with the `versions`. 
![Selection_002](https://user-images.githubusercontent.com/8020943/54710461-03809e00-4b48-11e9-9cb4-ecb5beb861c1.png)

**Comment by prisae** [comment]
That is interesting. I never figured out, how to distinguish between a Notebook kernel and an IPython/Qt kernel. If you do the same in an IPython terminal, what will happen?

**Comment by prisae** [comment]
(The discussion around this problem is here: https://github.com/simpeg/simpeg/pull/698).

**Comment by banesullivan** [comment]
As a developer, you don't have to do anything besides implement a method called `_repr_htm_` that returns a string of your HTML representation - Jupyter handles figuring out that it should be used. Ipython/Qt would call the `__repr__` method by default

**Comment by banesullivan** [comment]
@prisae - if you want an HTML representation in notebooks and a plain text representation in IPython/Qt then simple make a version class object that implements the two `__repr__` and `_repr_html_` methods

**Comment by prisae** [comment]
I don't think it is that easy. See this screenshot from IPython: 
![Selection_001](https://user-images.githubusercontent.com/8020943/54711172-9c63e900-4b49-11e9-8866-921e3f2cca75.png)

**Comment by banesullivan** [comment]
What happens if you call `repr(examples.load_rectilinear)`?

**Comment by banesullivan** [comment]
Also here's something to test:

```py

class Information(object):

    @property
    def n_cells(self):
        return 6

    @property
    def bounds(self):
        return (-1,1, -1,1, -1,1)

    def _get_attrs(self):
        """An internal helper for the representation methods"""
        attrs = []
        attrs.append(("N Cells", self.n_cells, "{}"))
        bds = self.bounds
        attrs.append(("X Bounds", (bds[0], bds[1]), "{:.3e}, {:.3e}"))
        attrs.append(("Y Bounds", (bds[2], bds[3]), "{:.3e}, {:.3e}"))
        attrs.append(("Z Bounds", (bds[4], bds[5]), "{:.3e}, {:.3e}"))
        return attrs


    def __repr__(self):
        fmt = "{} ({})\n".format(type(self).__name__, hex(id(self)))
        # now make a call on the object to get its attributes as a list of len 2 tuples
        row = "  {}:\t{}\n"
        for attr in self._get_attrs():
            try:
                fmt += row.format(attr[0], attr[2].format(*attr[1]))
            except:
                fmt += row.format(attr[0], attr[2].format(attr[1]))
        return fmt

    def _repr_html_(self):
        fmt = ""
        # HTML version
        fmt += "\n"
        fmt += "<table>\n"
        fmt += "<tr><th>{}</th><th>Information</th></tr>\n".format(type(self).__name__)
        row = "<tr><td>{}</td><td>{}</td></tr>\n"
        # now make a call on the object to get its attributes as a list of len 2 tuples
        for attr in self._get_attrs():
            try:
                fmt += row.format(attr[0], attr[2].format(*attr[1]))
            except:
                fmt += row.format(attr[0], attr[2].format(attr[1]))
        fmt += "</table>\n"
        fmt += "\n"
        return fmt


Information()
```

**Comment by prisae** [comment]
![Selection_001](https://user-images.githubusercontent.com/8020943/54711885-35473400-4b4b-11e9-9cf2-0f71f02b2e05.png)

**Comment by banesullivan** [comment]
That's a bug on my part in `vtki` will fix now... try the basic `Information` class shared above

**Comment by prisae** [comment]
![Selection_002](https://user-images.githubusercontent.com/8020943/54712081-b6063000-4b4b-11e9-9cf0-0ec1f3cc5a4a.png)

**Comment by banesullivan** [comment]
Went ahead and added in the bare bones for you @prisae: #143

**Comment by lheagy** [comment]
This is looking good and would be a great contribution! 

Just a couple questions at this early stage: 
- what do you mean be `N` in the first column? Would 'domain' be appropriate here?
- Visually, it might be more intuitive if the number of cells in each dimension is in the table rather than the first line

**Comment by rowanc1** [comment]
I think that it would be absolutely awesome to improve this print out. Starting down the road of useful (albeit at the start) static widgets is a really cool avenue to explore.

The print function was originally done by @dwfmarchant, back in 2014 or so! Probably in the first versions of the python implementation - lots of room for improvement. :)

**Comment by prisae** [comment]
This was fixed in #143 .
